File: exercises/exercise-1-testing-java/src/main/java/io/miragon/example/adapter/inbound/zeebe/newsletter/AbortRegistrationWorker.java
package io.miragon.example.adapter.inbound.zeebe.newsletter;

import io.camunda.client.annotation.JobWorker;
import io.camunda.client.annotation.Variable;
import io.miragon.example.adapter.process.generated.NewsletterSubscriptionProcessApi.TaskTypes;
import io.miragon.example.application.port.inbound.newsletter.AbortSubscriptionUseCase;
import io.miragon.example.domain.SubscriptionId;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

import java.util.UUID;

@Component
public class AbortRegistrationWorker {
	
	private static final Logger log = LoggerFactory.getLogger(AbortRegistrationWorker.class);
	
	private final AbortSubscriptionUseCase useCase;
	
	public AbortRegistrationWorker(AbortSubscriptionUseCase useCase) {
		this.useCase = useCase;
	}
	
	@JobWorker(type = TaskTypes.NEWSLETTER_ABORT_REGISTRATION)
	public void handle(@Variable String subscriptionId) {
		log.debug("Received job to abort registration for subscriptionId: {}", subscriptionId);
		useCase.abort(new SubscriptionId(UUID.fromString(subscriptionId)));
	}
	
}

File: exercises/exercise-1-testing-kotlin/src/main/kotlin/io/miragon/example/adapter/inbound/zeebe/newsletter/AbortRegistrationWorker.kt
package io.miragon.example.adapter.inbound.zeebe.newsletter

import io.miragon.example.adapter.process.generated.NewsletterSubscriptionProcessApi.TaskTypes
import io.miragon.example.application.port.inbound.newsletter.AbortSubscriptionUseCase
import io.miragon.example.domain.SubscriptionId
import io.camunda.client.annotation.JobWorker
import io.camunda.client.annotation.Variable
import mu.KotlinLogging
import org.springframework.stereotype.Component
import java.util.*

@Component
class AbortRegistrationWorker(
    private val useCase: AbortSubscriptionUseCase
) {

    private val log = KotlinLogging.logger {}

    @JobWorker(type = TaskTypes.ACTIVITY_ABORT_REGISTRATION)
    fun handle(@Variable subscriptionId: String) {
        log.debug { "Received job to abort registration for subscriptionId: $subscriptionId" }
        useCase.abort(SubscriptionId(UUID.fromString(subscriptionId)))
    }
}
